webrender/src/renderer.rs

```diff
@@ -4516,7 +4545,17 @@ impl<B: hal::Backend> Renderer<B> {
             &target.blits, render_tasks, draw_target, &content_origin,
         );
         #[cfg(not(feature = "gl"))]
-        self.device.begin_render_pass(transit_to_present && target.alpha_batch_containers.is_empty());
+        {
+            if target_kind == TargetKind::Main || !target.empty_without_batches() {
+                self.device.begin_render_pass(transit_to_present && target.alpha_batch_containers.is_empty());
```

Comment:
this reads a bit misplaced to me: by this moment we should have already done the clears and blits, both of which require the render pass. So why don't we make sure that the render pass starts earlier and here assume it's already set up?

Reply:
The clear above clears the entire texture (because the clear rect is none with gfx backends: https://github.com/szeged/webrender/pull/343/files#diff-f5062b694b9fe53fc1757ed483d577d9R4528) so we will use a clear load operation in this case, and we record that load operation in `clear_target` prior to render pass begin. So that clear must be outside of render pass. Because that clear is outside of a render pass, The blit is outside of a render pass as well in this case we don't force to blot with a shader and use the default blit call.